### DISCUSSION TITLE
Create a Recipe Activity Timeline and add a "last date made" property to track when a user makes a Recipe

### DISCUSSION THREAD

**Discussion by michael-genson** [mealie-recipes/mealie#1451] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Something I'm constantly going back and asking myself is "when's the last time I made this?" It would be great if we could track activities on a recipe such as timestamping every time I make it. It could also track other events like "User _ABC_ created this recipe" or "User _DEF_ modified this recipe". Those are currently covered by their individual `created_at` and `update_at` properties, but it doesn't track history or user (to my knowledge, anyway, and if it does it's not exposed). This could also play nice with meal plans which could have their own events.

**(Optional): Suggest A Solution**
This requires a few changes, though I think each is an incremental improvement, and not necessarily dependent on all of these being completed at once:
1. Extend the Recipe schema to include `last_made`
2. Add an "I made this" button on the Recipe page
3. Extend the Recipe schema to track events over time (with a new `Activity` class that stores the actor, timestamp, and activity details), or create a new `RecipeTimeline` child object that stores activities
4. Add API routes for the Recipe timeline
5. Integrate Create, Update, Last Made, and Meal Plans as their own Activities
6. Expose the Recipe timeline to the frontend, probably on its own page (this is likely the most work out of all the changes)

One example of this I've come across that does this well is Freshdesk (a business-to-user helpdesk):
[General usage](https://support.freshdesk.com/en/support/solutions/articles/50000001172-activity-api)
[Actual API documentation](https://developers.freshdesk.com/api/#publish_omnichannel_activity)

**Additional Information**
- [X] If the feature is accepted I'm willing to submit a PR to provide this feature

If accepted I'm happy to work on this, though I'll probably need a hint for the database changes, and I'm not sure I could do the last task with the frontend.

**Comment by hay-kot** [comment]
Yeah, I think this is a good addition, I think limiting the scope to just include the times you've made the recipe would be appropriate to start with. We can use a string enumeration to categorize events in the database like we do for some of the other things we track, that way we can always extend it as needed.

**Reply by michael-genson** [reply]
> We can use a string enumeration to categorize events in the database

Hmm, when you put it like that, I feel like this might play nicely with a new event listener using the new event flow. The "I made this" button could trigger a recipe update event and we could listen for recipe updates. Maybe. I'll play around with it
